packages/flutter_goldens/lib/flutter_goldens.dart

```diff
@@ -9,79 +9,153 @@ import 'package:file/file.dart';
 Future<void> main(FutureOr<void> testMain()) async {
-  goldenFileComparator = await FlutterGoldenFileComparator.fromDefaultComparator();
+  const Platform platform = LocalPlatform();
+  goldenFileComparator = _testingWithSkiaGold(platform)
+    ? await FlutterSkiaGoldFileComparator.fromDefaultComparator()
+    : platform.isLinux
+      ? await FlutterGoldensRepositoryFileComparator.fromDefaultComparator()
+      : await FlutterDummyGoldenFileComparator.fromDefaultComparator();
   await testMain();
 }
 
-/// A golden file comparator specific to the `flutter/flutter` repository.
-///
-/// Within the https://github.com/flutter/flutter repository, it's important
-/// not to check-in binaries in order to keep the size of the repository to a
-/// minimum. To satisfy this requirement, this comparator retrieves the golden
-/// files from a sibling repository, `flutter/goldens`.
-///
-/// This comparator will locally clone the `flutter/goldens` repository into
-/// the `$FLUTTER_ROOT/bin/cache/pkg/goldens` folder, then perform the comparison against
-/// the files therein.
-class FlutterGoldenFileComparator implements GoldenFileComparator {
+/// Abstract base class golden file comparator specific to the `flutter/flutter`
+/// repository.
+abstract class FlutterGoldenFileComparator implements GoldenFileComparator {
   /// Creates a [FlutterGoldenFileComparator] that will resolve golden file
   /// URIs relative to the specified [basedir].
   ///
-  /// The [fs] parameter exists for testing purposes only.
+  /// The [fs] and [platform] parameters are only for use in tests, where the
+  /// default file system and platform can be replaced by mock instances.
   @visibleForTesting
   FlutterGoldenFileComparator(
     this.basedir, {
     this.fs = const LocalFileSystem(),
-  });
+    this.platform = const LocalPlatform(),
+  }) : assert(basedir != null);
 
-  /// The directory to which golden file URIs will be resolved in [compare] and [update].
+  /// The directory to which golden file URIs will be resolved in [compare] and
+  /// [update].
   final Uri basedir;
 
   /// The file system used to perform file access.
   @visibleForTesting
   final FileSystem fs;
 
-  /// Creates a new [FlutterGoldenFileComparator] that mirrors the relative
-  /// path resolution of the default [goldenFileComparator].
+  /// A wrapper for the [dart:io.Platform] API.
+  @visibleForTesting
+  final Platform platform;
+
+  @override
+  @protected
+  @visibleForTesting
+  Future<void> update(Uri golden, Uint8List imageBytes) async {
+    final File goldenFile = getGoldenFile(golden);
+    await goldenFile.parent.create(recursive: true);
+    await goldenFile.writeAsBytes(imageBytes, flush: true);
+  }
+
+  /// Modifies the given [Uri] to incorporate the current test library as a
+  /// prefix in the golden file name.
+  @override
+  @protected
+  @visibleForTesting
+  Uri getTestUri(Uri key, int version) {
+    return key;
```

Comment:
I wonder if the default should be to mingle the version into the key to avoid some of the duplication of that logic?